File: app/src/main/java/com/happs/ximand/ringcontrol/model/repository/Repository.java
package com.happs.ximand.ringcontrol.model.repository;

import com.happs.ximand.ringcontrol.model.specification.SqlSpecification;

import java.util.List;

public interface Repository<T> {
    void add(T item);

    void update(T item);

    void remove(T item);

    List<T> query(SqlSpecification sqlSpecification);
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/repository/Repository.kt
package com.happs.ximand.ringcontrol.model.repository

import com.happs.ximand.ringcontrol.model.specification.SqlSpecification

interface Repository<T> {
    fun add(item: T)
    fun update(item: T)
    fun remove(item: T)
    fun query(sqlSpecification: SqlSpecification): List<T>
}